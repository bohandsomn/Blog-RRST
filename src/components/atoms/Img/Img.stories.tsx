import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import Img from './Img'
import AtomProvider from '../../../packages/atom/Provider'
import STYLES from '../../../styles'

export default {
  title: 'Atom/Img',
  component: Img,
} as ComponentMeta<typeof Img>

const Template: ComponentStory<typeof Img> = (args) => {
    const classNameList: (keyof typeof STYLES)[] = ['border-radius-50%', 'background-color-button-shadow']
    return (
        <AtomProvider styles={STYLES}>
            <Img className={classNameList.join(' ')} {...args} />
        </AtomProvider>
    )
}

export const Original = Template.bind({})
Original.args = {
    width: 200,
    height: 200,
}

export const Post = Template.bind({})
Post.args = {
    width: 75,
    height: 75,
}

export const Preview = Template.bind({})
Preview.args = {
    width: 60,
    height: 60,
}

export const Message = Template.bind({})
Message.args = {
    width: 20,
    height: 20,
}