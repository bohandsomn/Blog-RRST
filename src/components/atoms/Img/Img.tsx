import React from 'react'
import Atom from '@/packages/atom'
import useAppStyles from '../../../hooks/useAppStyles'

export const Img = Atom.create('img')

type SeoProps = Omit<NonNullable<typeof Img.defaultProps>, 'alt' | 'title' | 'src'> & {
    alt: string
    title: string
    src: string
}

export const OriginalImg = React.forwardRef(({className, ...props}: SeoProps, ref) => {
    const localClassName = useAppStyles('display-flex', 'border-radius-50%', 'background-color-button-shadow', className as undefined)
    return (
        <Img className={localClassName} ref={ref} width={200} height={200} {...props} />
    )
}) 

export const PostImg = React.forwardRef(({className, ...props}: SeoProps, ref) => {
    const localClassName = useAppStyles('display-flex', 'border-radius-50%', 'background-color-button-shadow', className as undefined)
    return (
        <Img className={localClassName} ref={ref} width={75} height={75} {...props} />
    )
}) 

export const PreviewImg = React.forwardRef(({className, ...props}: SeoProps, ref) => {
    const localClassName = useAppStyles('display-flex', 'border-radius-50%', 'background-color-button-shadow', className as undefined)
    return (
        <Img className={localClassName} ref={ref} width={60} height={60} {...props} />
    )
}) 

export const MessageImg = React.forwardRef(({className, ...props}: SeoProps, ref) => {
    const localClassName = useAppStyles('display-flex', 'border-radius-50%', 'background-color-button-shadow', className as undefined)
    return (
        <Img className={localClassName} ref={ref} width={20} height={20} {...props} />
    )
})
