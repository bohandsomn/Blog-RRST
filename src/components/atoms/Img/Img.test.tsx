import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Img, { OriginalImg, PostImg, PreviewImg, MessageImg } from './Img'

describe('Atom Img', () => {
    test('should renders the element', () => {
        render(<Img />)
        const element = screen.getByRole('img')
        expect(element).toBeInTheDocument()
    })

    test('OriginalImg should not differ from the previous snapshot', () => {
        render(<OriginalImg />)
        const element = screen.getByRole('img')
        expect(element).toMatchSnapshot()
    })

    test('PostImg should not differ from the previous snapshot', () => {
        render(<PostImg />)
        const element = screen.getByRole('img')
        expect(element).toMatchSnapshot()
    })

    test('PreviewImg should not differ from the previous snapshot', () => {
        render(<PreviewImg />)
        const element = screen.getByRole('img')
        expect(element).toMatchSnapshot()
    })

    test('MessageImg should not differ from the previous snapshot', () => {
        render(<MessageImg />)
        const element = screen.getByRole('img')
        expect(element).toMatchSnapshot()
    })
})