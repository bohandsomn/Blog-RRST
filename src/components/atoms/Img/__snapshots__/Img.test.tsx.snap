// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom Img MessageImg should not differ from the previous snapshot 1`] = `
<img
  class="sc-bcXHqe"
  height="20"
  width="20"
/>
`;

exports[`Atom Img OriginalImg should not differ from the previous snapshot 1`] = `
<img
  class="sc-bcXHqe"
  height="200"
  width="200"
/>
`;

exports[`Atom Img PostImg should not differ from the previous snapshot 1`] = `
<img
  class="sc-bcXHqe"
  height="75"
  width="75"
/>
`;

exports[`Atom Img PreviewImg should not differ from the previous snapshot 1`] = `
<img
  class="sc-bcXHqe"
  height="60"
  width="60"
/>
`;
